Controller/Adminhtml/Menu/Delete.php

```diff
@@ -53,19 +81,22 @@ public function __construct(
     /**
      * Dispatch request
      *
-     * @return \Magento\Framework\Controller\ResultInterface|ResponseInterface
-     * @throws \Magento\Framework\Exception\NotFoundException
+     * @return ResultInterface|ResponseInterface
+     * @throws NotFoundException
      */
     public function execute()
     {
-        $id = $this->getRequest()->getParam('id');
+        $menuId = (int) $this->getRequest()->getParam('id');
 
         try {
-            $menu = $this->menuRepository->getById($id);
-            $this->menuRepository->deleteById($id);
+            $menu = $this->menuRepository->getById($menuId);
+            $this->menuRepository->deleteById($menuId);
```

Comment:
I think we could use `$this->menuRepository->delete($menu)`, this way we avoid loading a menu twice

Reply:
Done